lib/oauth2/strategy/base.rb

```diff
@@ -11,6 +11,14 @@ def initialize(client)
       def client_params
         {'client_id' => @client.id, 'client_secret' => @client.secret}
       end
+
+      # Returns the Authorization header value for Basic Authentication
+      #
+      # @param [String] The client ID
+      # @param [String] the client secret
+      def authorization(client_id, client_secret)
+        'Basic ' + Base64.encode64(client_id + ':' + client_secret).gsub("\n", '')
```

Comment:
@divoxx I don't think that is correct in the case of the `Authorization` header, as it is Base64 encoded. I wasn't sure so I tested to see what `curl` does here. Furthermore, the RFC does not specify any encoding other than Base64: https://tools.ietf.org/html/rfc2617#section-2.